.htaccess

```diff
@@ -5,6 +5,11 @@
 
     ### =============== Legacy redirects ===============
 
+    # Redirect non-languaged minor versions to en, e.g. /3.1/class-DataObject.html to /en/3/class-DataObject.html
+    RewriteCond %{REQUEST_URI} ^(.*)(\d)\.\d/(.*)$ [NC]
+    RewriteRule ^(.*)$ %1en/%2/%3 [L,R=301]
```

Comment:
The above pattern is pretty risky, shall we move the decimal digit check to the rewriterule condition instead?

RewriteRule excludes the base url, so you don't need the leading (.*)